net/ftpproxy/src/opnsense/mvc/app/models/OPNsense/FtpProxy/FtpProxy.php

```diff
@@ -0,0 +1,89 @@
+<?php
+
+/**
+ *    Copyright (C) 2016 EURO-LOG AG
+ *
+ *    All rights reserved.
+ *
+ *    Redistribution and use in source and binary forms, with or without
+ *    modification, are permitted provided that the following conditions are met:
+ *
+ *    1. Redistributions of source code must retain the above copyright notice,
+ *       this list of conditions and the following disclaimer.
+ *
+ *    2. Redistributions in binary form must reproduce the above copyright
+ *       notice, this list of conditions and the following disclaimer in the
+ *       documentation and/or other materials provided with the distribution.
+ *
+ *    THIS SOFTWARE IS PROVIDED ``AS IS'' AND ANY EXPRESS OR IMPLIED WARRANTIES,
+ *    INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY
+ *    AND FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+ *    AUTHOR BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY,
+ *    OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ *    SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ *    INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ *    CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ *    ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ *    POSSIBILITY OF SUCH DAMAGE.
+ *
+ */
+
+namespace OPNsense\FtpProxy;
+
+use OPNsense\Base\BaseModel;
+
+/**
+ * Class FtpProxy
+ * @package OPNsense\FtpProxy
+ */
+class FtpProxy extends BaseModel
+{
+	/**
+	 * map config to ftp-proxy flags
+	 * and set default values
+	 * @param $node configuration
+	 * @return string
+	 */
+	public function configToFlags($node)
```

Comment:
@fbrendel sorry for the late response, it has been busy. Your right, splitting the calls isn't a big thing, it only helps people using the api to control the workflow better. I don't want to insist here, but I do think its cleaner :)

For the parameter mapping I would use the template like your doing now and add a custom version for your rc(8) script (a variant of /etc/rc.d/ftp-proxy named for example opn-ftp-proxy)

In that case you could use the pid files for the daemons spawned and add an extra command using the "extra_commands" directive, which checks for new/removed daemons and start/stops them. should be relatively easy todo, ditches the need for FtpProxy.sh and the additional php code, the configd calls can integrate into the normal rc(8) system.